strings/remove duplicates.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:test/test.dart';
+
+// function to remove duplicate in string
+String removeDups(String s) {
+  int j = 0;
+  String l = '';
+  int i = 0;
+  var check = 0;
+  for (i = 0; i < s.length; i++) {
+    check = 0;
+    var c = s[i];
+    if (l.length != 0) {
+      for (j = 0; j < l.length; j++) {
+        if (c == l[j]) {
+          check = 1;
+          break;
+        } else
+          check = 0;
+      }
+      if (check == 0) l += c;
+    } else
+      l += c;
+  }
```

Comment:
If I am not wrong then this has a time complexity of O(N^2), right?
We can make this better with a space-time trade-off where we can use HashMap or Set data structure to track the elements already encountered.
This will take the time complexity to O(N) and space O(N)
or you can use an integer array of size 256 to track the elements already encountered. where each character will be stored in its Ascii value index.
This would work in a single pass through the string also

Reply:
Is this already resolved?

Reply:
@Parowicz I have made  the suggested algorithmic changes,please review and merge.